File: 20200215Python-China/0448. Find All Numbers Disappeared in an Array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        k=0
        while k<len(nums):
            if nums[k]==k+1 or nums[k]==nums[nums[k]-1]:
                k+=1
            else:
                tmp=nums[k]
                nums[k],nums[tmp-1]=nums[tmp-1],nums[k]
        
        rep=[]
        for k in range(len(nums)):
            if nums[k]!=k+1:
                rep.append(k+1)
        return rep
            

File: archives/C#/0448. Find All Numbers Disappeared in an Array.cs
public class Solution {
    public IList<int> FindDisappearedNumbers(int[] nums) {
        int i=0;
        while(i<nums.Length){
            if (nums[i]-1!=i && nums[nums[i]-1]!=nums[i]){
                int tmp=nums[i];
                nums[i]=nums[tmp-1];
                nums[tmp-1]=tmp;
            }
            else{
                i+=1;
            }
        }
        IList<int> rep=new List<int>();
        for(int j=0;j<nums.Length;j++){
            if (nums[j]-1!=j){
                rep.Add(j+1);
            }
        }
        return rep;
    }
}
